Add community page.

fwalch opened the pull request:
The idea is to link to this page from https://github.com/neovim/neovim#community, https://github.com/neovim/neovim/wiki/Communicating, and https://github.com/neovim/neovim/wiki/Contributing#give-support.

Speaking of https://github.com/neovim/neovim/wiki/Communicating, should the code of conduct be linked on this page as well? Is "Support" a good title, or should it be called e.g. "Community" instead?

fwalch commented:
The idea is to link to this from https://github.com/neovim/neovim#community, https://github.com/neovim/neovim/wiki/Communicating, and https://github.com/neovim/neovim/wiki/Contributing#give-support.

Speaking of https://github.com/neovim/neovim/wiki/Communicating, should the code of conduct be linked on that page as well? Is "Support" a good title, or should it be called e.g. "Community" instead?

Pyrohh commented:
> Is "Support" a good title, or should it be called e.g. "Community" instead?

"Community" is better IMO, as wanting support isn't necessarily the reason someone wants to interact with a community such as this. Anyways, I really wish that we had some unified place to go for information. We have the vim differences wiki page, and (hopefully soon) the same thing for docs, which must be manually synchronized.

I think what you're doing is better than the current setup, but it still means users are being directed to a different domain just to see such a small page. Perhaps we could just have everything written in markdown, and then periodically (at least when something's changed) regenerate it for use in specific wiki pages and the website, and have @marvim automatically push everything out. Just an idea.

fwalch commented:
> We have the vim differences wiki page, and (hopefully soon) the same thing for docs, which must be manually synchronized.

I thought that once it was in the docs, the wiki page would go away entirely.

In general, I think users will check out the website first, so IMO user-stuff should live there, with the Github Wiki as a resource for (Neovim/plugin/..) developers.

Pyrohh commented:
> I thought that once it was in the docs, the wiki page would go away entirely.

For each change, it usually has a link to the PR that made it (in case users don't know why it was removed and want to see the discussion). Does nvim support hyperlinks? If so, then I imagine that the built-in docs could obviate the wiki page.

> In general, I think users will check out the website first, so IMO user-stuff should live there, with the Github Wiki as a resource for (Neovim/plugin/..) developers. That way we could probably avoid most duplication.

I suppose that makes sense, but what do you mean by "user stuff"? Generated docs?

fwalch commented:
> I suppose that makes sense, but what do you mean by "user stuff"? Generated docs?

That is a good question, actually :grin:
In the Wiki, there's currently:
 * introduction & FAQ: should maybe be moved into `:help` at some point.
 * documentation: already only links to website.
 * installing: should probably go to the website as well, at least for released versions (as soon as we have one). Installation instructions for unstable versions can still live in the wiki.

I admit that I haven't thought all of this through entirely, so please tell me your opinion where such "stuff" should be placed. It only bothered me that the "community links" are in so many places.

Pyrohh commented:
> introduction & FAQ: should probably be moved to website/:help at some point.
> documentation: already only links to website.
> installing: should probably go to the website as well, at least for released versions (as soon as we have one). Installation instructions for unstable versions can still live in the wiki.
> I admit that I haven't thought all of this through entirely, so please tell me your opinion where such "stuff" should be placed. It only bothered me that the "community links" are in so many places.

I know you said you haven't thought this through, but that sounds pretty solid to me. I think the following distinctions should be made in all cases as to have a clear seperation between the three:

- Website: obvious user facing things
- Wiki: only developer related things
- Built-in docs: everything else (that is to say that we should remove things like `usr_90.txt`)

justinmk commented:
Moving things to the website means it's a pain to update them. Wiki is far more amenable to people making small edits, as has been seen with the install page. Those people would not have bothered if they were required to make a PR.

Wiki should not be only for developer-related things.

fwalch commented:
> Wiki is far more amenable to people making small edits, as has been seen with the install page.

True, but let's put it like this: as long as user-facing things like that change that often, Neovim is not ready to be used by everyone (i.e. it's still experimental). Installation instructions for released version, on the other hand, won't change (only for new versions). Instructions for "bleeding edge" versions can still remain in the Wiki.

Pyrohh commented:
> Moving things to the website means it's a pain to update them. Wiki is far more amenable to people making small edits, as has been seen with the install page. Those people would not have bothered if they were required to make a PR.

I meant this when I said developer only things, but you make a good point in that it would be a pain to update. Still, installation instructions for stable versions are most likely not going to change much, as @fwalch said.

> Wiki should not be only for developer-related things.

Why? It currently is almost exclusively developer only things (or at least contributor only), and it makes more sense IMO as many end users are never going to go to GitHub once they can get Neovim from their package manager.

fwalch commented:
@justinmk Regardless of whether installation instructions should move to the website, what do you think of the PR itself?

Pyrohh commented on community.html:
```diff
@@ -0,0 +1,32 @@
+      </p>
+
+      <ul>
+        <li>Mailing list: <a href="https://groups.google.com/forum/#!forum/neovim">Neovim Google group</a></li>
```
Gitter seems to be significantly more active than anything else here, so perhaps it should be at the top.

Pyrohh commented:
I'm clearly not @justinmk, but I think this is a pretty good idea. The Google group being in the top bar is misleading IMO, as it's probably not the best way to get in touch with the Neovim community.

justinmk commented on _data/nav.yml:
```diff
@@ -4,11 +4,11 @@
-  url: http://twitter.com/Neovim
-- title: Bountysource
+  url: https://twitter.com/neovim
+- title: Donations
```
I hope we never have the word "donate" anywhere in our materials. It's really tacky. If people want to donate they will. How about "Support". Or "$upport" if you feel peachy :)

Pyrohh replied:
Oh no... please not the second one :)

On April 2, 2015 7:35:22 PM EDT, "Justin M. Keyes" <notifications@github.com> wrote:
>>  - title: Twitter
>> -  url: http://twitter.com/Neovim
>> -- title: Bountysource
>> +  url: https://twitter.com/neovim
>> +- title: Donations
>
>I hope we never have the word "donate" anywhere in our materials. It's
>really tacky. If people want to donate they will. How about "Support".
>Or "$upport" if you feel peachy :)
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/neovim/neovim.github.io/pull/91/files#r27706783

justinmk replied:
That would be more tacky :)

justinmk replied:
Actually, let's just remove this link. It has no need to be in the ever-present top navigation. The bitcoin/donate area below the fold on the mainpage is good enough. Anyone who wants to donate will have no trouble finding that.

justinmk commented:
@Pyrohh @fwalch community page is worth having, although later it might be better to move it to the front page.

fwalch commented:
Removed the bountysource link and moved Gitter to the top.

Pyrohh commented:
LGTM, although maybe I should actually see what it looks like rendered... if you already have this set up @fwalch, could you take a screenshot? I feel pretty silly, but I spent an hour yesterday trying to get the site up and running locally, but I couldn't get bundler to install my bundle or gem or whatever you call it :grin:

fwalch commented:
Fixed some error with a `<code>` tag; here's a screenshot:

![gui](https://cloud.githubusercontent.com/assets/339435/6994283/1f8b0d2e-db1c-11e4-9de7-eff084d99025.jpg)

FWIW, here's how I think I set up bundler on my Arch system:

 * Install ruby via pacman
 * `gem install bundler`
 * Adjust PATH to contain `~/.gem/ruby/2.2.0/bin` (bit unfortunate that this path will be out-of-date if a new ruby version comes out, but it won't break -- just noticed I used the bundler installed into the 2.1.0 subdirectory all the time)
 * Open a shell and execute:
```
$ cd neovim.github.io
$ mkdir .bundle
$ cat > .bundle/config
---
BUNDLE_PATH: .bundle
BUNDLE_DISABLE_SHARED_GEMS: '1'
$ bundle
$ bundle exec jekyll serve
```
* Visit the website at 0.0.0.0:4000

Pyrohh commented on community.html:
```diff
@@ -0,0 +1,32 @@
+      </ul>
+
+      <p>
+      If you want to report a bug, please use <a href="https://github.com/neovim/neovim/issues">Neovim's issue tracker on Github</a>.
```
`Github` should be `GitHub`, and perhaps instead of:
```
If you want to report a bug, ...
```
it could be
```
If you want to report a bug or suggestion, ...
```
or more generally
```
If you want to report an issue, ...
```

Pyrohh commented:
Thanks for providing the instructions, but unfortunately they didn't work for me: jekyll complained about a missing JS interpreter, and I have no idea what I'm supposed to use. No worries though, as I think the screenshot provided is enough. Besides the issues I posted, LGTM.

jdavis commented:
:+1: I like this. I've been wanting to change the nav links across the top for awhile now. The double word Google Group and long Bountysource were bugging me.

Is there any reason why the Twitter isn't removed since it is included in the Community tab? The Twitter link certainly is important. What do you guys think?

Pyrohh commented:
> Is there any reason why the Twitter link isn't removed since it is included in the Community tab? The Twitter link certainly is important but... What do you guys think?

If we're cutting down on the amount of items in that bar, then I think it makes sense. The twitter account is a good way to keep up with various happenings in Neovim (not just necessarily development, but community as well).

fwalch commented:
Moved the Twitter link to the community page and rewrote the introductory sentence.

![gui](https://cloud.githubusercontent.com/assets/339435/6996857/e4c05732-dbaa-11e4-8ea6-44b8b4e313ee.jpg)

jdavis commented:
That looks really great, @fwalch. I like the addition of this page.

fwalch commented:
Thanks! Forgot to add that I changed the stylesheet to show the link of the currently active page as bold.

jdavis commented:
Yeah, I saw that. It looks good. Do you think we should get more feedback before merging?

Pyrohh commented:
I think this change is very straightforward, but maybe see what @justinmk thinks?

fwalch commented:
Can this be merged? :-)
